workspaces/rbac/plugins/rbac-backend/src/service/permission-definition-routes.ts

```diff
@@ -0,0 +1,171 @@
+} from '@backstage/backend-plugin-api';
+import { ExtendablePluginIdProvider } from './extendable-id-provider';
+import {
+  ConflictError,
+  NotAllowedError,
+  NotFoundError,
+} from '@backstage/errors';
+
+// todo: do we need separated permissions to set up extra plugin ids  ?
+export function createPermissionDefinitionRoutes(
+  pluginPermMetaData: PluginPermissionMetadataCollector,
+  pluginIdProvider: ExtendablePluginIdProvider,
+  extraPluginsIdStorage: PermissionDependentPluginStore,
+  deps: {
+    auth: AuthService;
+    httpAuth: HttpAuthService;
+    auditor: AuditorService;
+    permissions: PermissionsService;
+  },
+) {
+  const router = Router();
+
+  const { auth, auditor } = deps;
+
+  router.get(
+    '/plugins/policies',
+    logAuditorEvent(auditor),
+    async (request, response) => {
+      await authorizeConditional(request, policyEntityReadPermission, deps);
+
+      const body = await pluginPermMetaData.getPluginPolicies(auth);
+
+      response.json(body);
+    },
+  );
+
+  router.get(
+    '/plugins/condition-rules',
+    logAuditorEvent(auditor),
+    async (request, response) => {
+      await authorizeConditional(request, policyEntityReadPermission, deps);
+
+      const body = await pluginPermMetaData.getPluginConditionRules(auth);
+
+      response.json(body);
+    },
+  );
+
+  router.get(
+    '/plugins/id',
+    logAuditorEvent(auditor),
+    async (request, response) => {
+      await authorizeConditional(request, policyEntityReadPermission, deps);
+
+      const actualPluginIds = await pluginIdProvider.getPluginIds();
+      response.status(200).json(pluginIdsToResponse(actualPluginIds));
+    },
+  );
+
+  router.post(
+    '/plugins/id',
+    logAuditorEvent(auditor),
+    async (request, response) => {
+      await authorizeConditional(request, policyEntityCreatePermission, deps);
+      const pluginIds: PermissionDependentPluginList = request.body;
+      // todo validate pluginIds object
+      const pluginDtos = permissionDependentPluginListToDTO(pluginIds);
+
+      let actualPluginIds = await pluginIdProvider.getPluginIds();
+      const conflictedIds = pluginIds.ids.filter(id =>
+        actualPluginIds.includes(id),
+      );
+      if (conflictedIds.length > 0) {
+        throw new ConflictError(
+          `Plugin IDs ${JSON.stringify(conflictedIds)} already exist in the system. Please use a different set of plugin ids.`,
+        );
+      }
+      await extraPluginsIdStorage.addPlugins(pluginDtos);
+      response.locals.meta = pluginIds;
+
+      actualPluginIds = await pluginIdProvider.getPluginIds();
+      response.status(201).json(pluginIdsToResponse(actualPluginIds));
+    },
+  );
+
+  router.delete(
+    '/plugins/id',
+    logAuditorEvent(auditor),
+    async (request, response) => {
+      await authorizeConditional(request, policyEntityDeletePermission, deps);
+      const pluginIds: PermissionDependentPluginList = request.body;
+      // todo validate pluginIds object
```

Comment:
Were you still planning on completing this todo?

Reply:
Done